+From: Troy Heber <troyh@debian.org>
Date: Tue, 08 Nov 2005 15:09:57 -0600
Subject: Avoid draining entropy pool

Changing the default behavior to not use apr_generate_random_bytes(), Debian use
the system APR which is configured to use /dev/random. In cases where the
entropy pool is drained svn commands can block waiting. Moving to the fallback
of using time. (Closes: #285708, #298822)
 
@DPATCH@
Index: subversion-1.7.0/subversion/libsvn_ra_svn/cram.c
===================================================================
--- subversion-1.7.0.orig/subversion/libsvn_ra_svn/cram.c	2011-10-20 20:47:31.000000000 +0200
+++ subversion-1.7.0/subversion/libsvn_ra_svn/cram.c	2011-10-20 21:55:46.000000000 +0200
@@ -123,12 +123,8 @@
  * should prevent replay attacks in practice. */
 static apr_status_t make_nonce(apr_uint64_t *nonce)
 {
-#if APR_HAS_RANDOM
-  return apr_generate_random_bytes((unsigned char *) nonce, sizeof(*nonce));
-#else
   *nonce = apr_time_now();
   return APR_SUCCESS;
-#endif
 }
 
 svn_error_t *svn_ra_svn_cram_server(svn_ra_svn_conn_t *conn, apr_pool_t *pool,
